lib/ecto_diff.ex

```diff
@@ -367,7 +367,47 @@ defmodule EctoDiff do
     end
   end
 
-  defp diff_association(previous, current, %{cardinality: :many, field: field, related: struct}, acc, opts) do
+  defp diff_association(
+         previous,
+         current,
+         %{cardinality: :many, through: [through, assoc_field], owner: owner},
+         acc,
+         opts
+       ) do
+    assoc_diff =
+      acc
+      |> Keyword.get_values(through)
+      |> List.flatten()
+      |> Enum.map(& &1[:changes][assoc_field])
+
+    diff =
+      case assoc_diff do
+        [_ | _] = diff ->
+          List.flatten(diff)
+
+        _ ->
```

Comment:
I don't quite understand what is expected value here... perhaps an empty list?

Reply:
If the `through` key is present, but the `assoc_field` key is not, the return could be `[nil]`, so `_ ->` was to handle both `[] || [nil]`. I have changed how we handle this so it can expect only empty lists.